tidy3d/components/viz.py

```diff
@@ -343,3 +345,28 @@ class SimulationViewerInjector {
     """
 
     return display(HTML(html_code))
+
+
+def set_default_labels_and_title(
+    axis_labels: tuple[str, str], axis: Axis, position: float, ax: Ax, plot_units: LengthUnit = None
+) -> Ax:
+    """Adds axis labels and title to plots involving spatial dimensions.
+    When the ``plot_units`` are specified, the plot axes are scaled, and
+    the title and axis labels include the desired units.
+    """
+    xlabel = axis_labels[0]
+    ylabel = axis_labels[1]
+    if plot_units is not None:
+        ax.set_xlabel(f"{xlabel} ({plot_units})")
+        ax.set_ylabel(f"{ylabel} ({plot_units})")
+        # Formatter to help plot in arbitrary units
+        scale_factor = UnitScaling[plot_units]
```

Comment:
should we raise a custom `KeyError` if `plot_units` doesn't exist in UnitScaling?